__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -1970,6 +1970,46 @@ describe('mintTokens', () => {
     const tokenBalance4 = await hWallet.getBalance(tokenUid);
     const expectedAmount4 = expectedAmount2 + 100;
     expect(tokenBalance4[0]).toHaveProperty('balance.unlocked', expectedAmount4);
+
+    // Delegate mint back to wallet 1
+    const delegateResponse = await hWallet2.delegateAuthority(tokenUid, 'mint', address0);
+    expect(delegateResponse.hash).toBeDefined();
+    await waitForTxReceived(hWallet, delegateResponse.hash);
+    await waitForTxReceived(hWallet2, delegateResponse.hash);
+
+    const mintResponse5 = await hWallet.mintTokens(tokenUid, 100, { data: ['foobar'] });
+    expect(mintResponse5.hash).toBeDefined();
+    await waitForTxReceived(hWallet, mintResponse5.hash);
+
+    // Validating there is a correct reference to the custom token
+    expect(mintResponse5).toHaveProperty('tokens.length', 1);
+    expect(mintResponse5.tokens[0]).toEqual(tokenUid);
+
+    // Validating custom token balance
+    const tokenBalance5 = await hWallet.getBalance(tokenUid);
+    const expectedAmount5 = expectedAmount4 + 100;
+    expect(tokenBalance5[0]).toHaveProperty('balance.unlocked', expectedAmount5);
+
+    const dataOutput5 = mintResponse5.outputs[mintResponse5.outputs.length - 1];
+    expect(dataOutput5).toHaveProperty('value', 1);
+    expect(dataOutput5).toHaveProperty('script', Buffer.from([6, 102, 111, 111, 98, 97, 114, 172]));
```

Comment:
Instead of doing this, what do you think about doing the following to test if the script is correct?

```
const data = 'foobar';
const dataScript = new ScriptData(data);
expect(dataOutput5).toHaveProperty('script', dataScript.createScript());
```